### DISCUSSION TITLE
Pipeflow error (district heating with circ_pump_pressure)

### DISCUSSION THREAD

**Discussion by amayboudellal** [e2nIEE/pandapipes#528] [Q&A] [answered]
Hello,

I'm building a district heating circular network with the following assets:
```
This pandapipes network includes the following parameter tables:
   - junction (688 elements)
   - junction_geodata (688 elements)
   - pipe (877 elements)
   - pipe_geodata (877 elements)
   - std_type (2 elements)
   - heat_exchanger (143 elements)
   - circ_pump_pressure (1 elements).
```

When trying to run a pipeflow, I get the following error message:
```
  File "/build_grid.py", line 563, in grid
    pp.pipeflow(net, mode='all')
  File "/.pyenv/versions/district_heating/lib/python3.8/site-packages/pandapipes/pipeflow.py", line 68, in pipeflow
    node_pit, branch_pit = initialize_pit(net, Junction.table_name())
  File "/.pyenv/versions/district_heating/lib/python3.8/site-packages/pandapipes/pipeflow_setup.py", line 333, in initialize_pit
    comp.create_pit_node_entries(net, pit["node"], node_name)
  File "/.pyenv/versions/district_heating/lib/python3.8/site-packages/pandapipes/component_models/circulation_pump_pressure_component.py", line 42, in create_pit_node_entries
    juncts_p, press_sum, number = _sum_by_group(circ_pump.to_junction.values,
  File "/.pyenv/versions/district_heating/lib/python3.8/site-packages/pandapipes/internals_toolbox.py", line 53, in _sum_by_group
    val[i] = val[i][order]
IndexError: index 0 is out of bounds for axis 0 with size 0
```

It might be hard to identify what is going on with so little information, but do you have an eventual guess?

**Comment by dlohmeier** [comment]
Hello @amayboudellal ,
thanks for letting us know about the issue. We are currently dealing with a few issues that came up concerning the circulation pump model, so this might be helpful for us as well. Could you give me some insight into the circulation pump settings (maybe just what the table looks like)? It definitely looks like the model is assuming that the pump is set out of service... such a case still needs to be caught as exception, we will consider this aspect in the next hotfix. But this wouldn't solve your problem, as it would just give a better output, but you still couldn't run a pipeflow. If you let me know what the circulation pump looks like, I will dive a little into the code.
Kind regards!

**Reply by amayboudellal** [reply]
Hello @dlohmeier,

Here is the circulation pump table:
```
  name  from_junction  to_junction  p_bar     t_k  plift_bar  in_service  type
0   16            340          684    4.0  343.15        1.0        True  auto
```

**Comment by dlohmeier** [comment, marked as answer]
Hello @amayboudellal ,
do you see the possibility to upgrade pandapipes to the latest version (0.8.4)? There the circulation pump model has already been updated. The columns were renamed, but this aspect should be considered automatically if you just load your network from JSON. If you just build your network on the fly, you would need a little bit of refactoring, but that shouldn't be too much.
Then we could see if that helps.

**Reply by amayboudellal** [reply]
Thank you, that solved my issue!

**Reply by dlohmeier** [reply]
Thank you for your feedback! :)
